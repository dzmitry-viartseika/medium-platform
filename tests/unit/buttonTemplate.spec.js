import { createLocalVue, shallowMount } from '@vue/test-utils';
import ButtonTemplate from '@/components/elements/buttonTemplate.vue';

const buttonSettings = {
  icon: false,
  buttonText: 'Sign in',
  buttonClickEvent: 'test',
  position: 'right',
};

describe('ButtonTemplate.vue', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(ButtonTemplate, {
      propsData: {
        buttonSettings,
      },
    });
    expect(wrapper.find('span').text()).toBe(buttonSettings.buttonText);
    const i = wrapper.find('i');
    expect(i.exists()).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });
});
