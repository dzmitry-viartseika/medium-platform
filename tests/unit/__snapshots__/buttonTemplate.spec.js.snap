// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTemplate.vue snapshot 1`] = `
<button class="app-button">
  <!----> <span>
      Sign in
    </span>
</button>
`;
